native/src/widget/grid.rs

```diff
@@ -0,0 +1,276 @@
+//! Write some text for your users to read.
+use crate::{
+    layout::{
+        flex::{self, Axis},
+        Limits, Node,
+    },
+    Align, Element, Hasher, Layout, Length, Point, Size, Widget,
+};
+use std::{any::TypeId, hash::Hash, iter};
+
+/// A container that distributes its contents in a grid.
+///
+/// # Example
+///
+/// ```
+/// # use iced_native::{renderer::Null, Element, Grid as NativeGrid, Text};
+/// # type Grid<'a> = NativeGrid<'a, (), Null>;
+/// Grid::with_children(
+///     vec![
+///         Element::from(Text::new("First row, first column")),
+///         Element::from(Text::new("First row, second column")),
+///         Element::from(Text::new("Second row, first column")),
+///         Element::from(Text::new("Second row, second column")),
+///     ],
+/// )
+/// .columns(2);
+/// ```
+#[allow(missing_debug_implementations)]
+pub struct Grid<'a, Message, Renderer> {
+    columns: Option<usize>,
+    column_width: Option<u16>,
```

Comment:
I would group this in an enum:

```rust
enum Strategy {
    Columns(usize),
    ColumnWidth(u16),
}
```

Removing the possibilities of "no strategy" or "two strategies" altogether.

Then, we can make `with_columns` and `with_column_width` the only constructor methods and remove `with_children` for now.

Reply:
Done.